src/content/blog/astro-290.mdx

```diff
@@ -0,0 +1,143 @@
+
+Enable support for view transitions in Astro 2.9 by adding the experimental flag to your config:
+
+```js
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  experimental: {
+    viewTransitions: true,
+  },
+});
+```
+
+This enables you to use the new APIs added.
+
+### &lt;ViewTransitions /&gt;
+
+This is a component which acts as the _router_ for transitions between pages. Add it to the `<head>` section of each individual page where transitions should occur _in the client_ as you navigate away to another page, instead of causing a full page browser refresh. 
+
+To enable support throughout your entire app (SPA mode), add the component in some common layout or component that targets the `<head>` of every page.
+
+**CommonHead.astro**
+
+```astro
+---
+import { ViewTransitions } from 'astro:transitions';
+---
+
+<meta charset="utf-8" />
+<title>{Astro.props.title}</title>
+<ViewTransitions />
+```
+
+With only this change, your app will now route completely in-client. You can then add transitions to individual elements using the `transition:animate` directive.
+
+### Animations
+
+Add `transition:animate` to any element to use Astro's built-in animations.
+
+```astro
+<header transition:animate="slide"></header>
+```
+
+In the above, Astro's `slide` animation will cause the `<header>` element to slide out to the left, and then slide in from the right when you navigate away from the page.
+
+You can also customize these animations using any CSS animation properties, for example, by specifying a duration:
+
+```astro
+---
+import { slide } from 'astro:transition';
+---
+
+<header transition:animate={slide({ duration: 200 })}></header>
+```
+
+### Continue learning
+
+Check out the [client-side routing docs](https://docs.astro.build/en/guides/client-side-routing/) to learn more.
+
+## Redirects
+
+This change moves the `redirects` configuration out of experimental status. If you were previously using experimental redirects, remove the following experimental flag:
+
+```js
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  experimental: {
+    redirects: true,
+  }
+});
+```
+
+If you have been waiting for stabilization before using redirects, wait no longer! You can now safely use redirects in your project. Check out [the docs on redirects](https://docs.astro.build/en/core-concepts/routing/#redirects) to learn how to use this built-in feature.
+
+In your configuration, add a map of old and new paths to redirect from and to:
+
+```js
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  redirects: {
+    '/old-page': '/new-page'
+  }
+});
+```
+
+## Hoisted script optimization
```

Comment:
Should we credit the author of the PR in this section? It'd be nice imo

Reply:
Yes, good point, I'll add that.